index.bs

```diff
@@ -2100,6 +2106,113 @@ Dictionary {{AudioTimestamp}} Members</h5>
+
+<h5 id="AudioRenderPerformance-attributes">
+Attributes</h5>
+
+<dl dfn-type=attribute dfn-for="AudioRenderPerformance">
+	: <dfn>onupdate</dfn>
+	::
+		An EventHandler for AudioRenderPerformanceEvent.
+</dl>
+
+<h5 id="AudioRenderPerformance-methods">
+Methods</h5>
+
+<dl dfn-type=method dfn-for="AudioRenderPerformance">
+	: <dfn>start(options)</dfn>
+	::
+		Starts performance-related data collection and analysis. This
+		will repeatedly dispatch an {{AudioRenderPerformanceEvent}} to
+		{{AudioRenderPerformance/onupdate}} EventHandler with the given
+		interval in {{AudioRenderPerformanceOptions}}.
+	
+	: <dfn>stop()</dfn>
+	::
+		Stops performance-related data collection and analysis. It also
+		stops the dispatching of {{AudioRenderPerformanceEvent}}.
+</dl>
+
+<h4 dictionary lt="audiorenderperformanceoptions" id="AudioRenderPerformanceOptions">
+{{AudioRenderPerformanceOptions}}</h4>
+
+The {{AudioRenderPerformanceOptions}} dictionary can be used to provide user
+options for an {{AudioRenderPerformance}}.
+
+<pre class="idl">
+dictionary AudioRenderPerformanceOptions {
+  	double updateInterval = 1;
+};
+</pre>
+
+<h5 id="dictionary-AudioRenderPerformanceOptions-members">
+Dictionary {{AudioRenderPerformanceOptions}} Members</h5>
+
+<dl dfn-type=dict-member dfn-for="AudioRenderPerformanceOptions">
+	: <dfn>updateInterval</dfn>
+	::
+		An interval (in second) that capacity/underrun values are
+		updated. A render capacity value is calculated per render
+		quantum, and each value will be collected over the specified
+		interval period. For example, if the renderer runs at a 48Khz
+		sample rate with 128 frames of render quantum over 1 second
+		interval, 375 capacity values will be collected.
+</dl>
+
+<h4 interface lt="audiorenderperformanceevent" id="AudioRenderPerformanceEvent">
+{{AudioRenderPerformanceEvent}}</h4>
+
+<pre class="idl">
+[Exposed=Window]
+interface AudioRenderPerformanceEvent {
+	constructor (DOMString type, double timestamp, double averageCapacity,
+	             double maxCapacity, double underrunRatio);
+  	readonly attribute double timestamp;
+  	readonly attribute double averageCapacity;
+  	readonly attribute double maxCapacity;
+  	readonly attribute double underrunRatio;
+};
+</pre>
+
+<h5 id="AudioRenderPerformanceEvent-attributes">
+Attributes</h5>
+
+<dl dfn-type=attribute dfn-for="AudioRenderPerformanceEvent">
+	: <dfn>timestamp</dfn>
+	::
+		{{BaseAudioContext/currentTime}} of {{AudioContext}} when the
+		event was fired.
+	: <dfn>averageCapacity</dfn>
+	::
+		An average of the collected raw render capacity values over the
+		specified interval. The precision is limited to 1/100th.
+	: <dfn>maxCapacity</dfn>
+	::
+		A maximum value from the collected raw render capacity values
+		over the specified interval. The precision is limited to
+		1/100th.
+	: <dfn>underrunRatio</dfn>
+	::
+		A ratio between the number of buffer underruns and the number
+		of render quanta. Where \(u\) is the number of underruns per
```

Comment:
It's probably obvious to us, but it's probably best to spec this in terms of "collection period", which are period of `udpateInterval` seconds, that are contiguous, not overlapping (since we have discrete blocks under the hood).  Then this underrun ratio is a property of a particular collection period, and it's clear when to "reset" the data collection.

When the system-level block size is not a divisor of the `updateInterval` * `AudioContext.sampleRate`, then we should spec it that it's `floor`ed (because we don't have the information for the very last block: it hasn't been rendered yet. It's more complicated if there is resampling or buffering somewhere in the browser, I guess.

Reply:
I might need a bit more clarifications here - how do you want to rephrase it? I liked the term "collection period", but do you want to use the term to explain this `underrun ratio` field? Or do we want to revise all the language above with this new term?

Reply:
I think a collection period can be the number of frames of a system-level audio callback, divided by the sample-rate the system-level audio stream runs at. An underrun is when the time it took to render the audio for a particular callback exceeded this collection period (e.g. it took 5ms to render 4ms of audio). The underrun ratio is then the number of underrun over the number of collection periods.

Reply:
There still is a concern about the system-level block size, but we'll get to it once we can agree on the general direction of the revision.

Reply:
I updated the text, so please take a look again.